### ISSUE TITLE
Pynamodb is not compatible with botocore 1.29.115

### ISSUE THREAD

**Issue by KirylKovaliov** [pynamodb/PynamoDB#1174]
In botocore 1.29.115 there is a breaking change which makes pynamodb incompatible with it.

It looks like botocore introduced mandatory 'endpoint_url' argument for the _convert_to_request_dict

How to reproduce:
`PynamoDbTable.scan()`

Error:
```
TypeError: _convert_to_request_dict() missing 1 required positional argument: 'endpoint_url'
Traceback (most recent call last):
  File "/var/lang/lib/python3.8/imp.py", line 234, in load_module
    return load_source(name, filename, file)
  File "/var/lang/lib/python3.8/imp.py", line 171, in load_source
    module = _load(spec)
  File "<frozen importlib._bootstrap>", line 702, in _load
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 843, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/var/task/http_lambda.py", line 19, in <module>
    from routes.validate import ValidateRoute
  File "/var/task/routes/validate.py", line 40, in <module>
    ip_black_listed=[item.ip for item in black_list_repo.get_many()],
  File "/opt/python/lib/python3.8/site-packages/licensingshared/repositories/dynamo.py", line 34, in get_many
    data = list(response)
  File "/opt/python/lib/python3.8/site-packages/pynamodb/pagination.py", line 194, in __next__
    self._get_next_page()
  File "/opt/python/lib/python3.8/site-packages/pynamodb/pagination.py", line 179, in _get_next_page
    page = next(self.page_iter)
  File "/opt/python/lib/python3.8/site-packages/pynamodb/pagination.py", line 115, in __next__
    page = self._operation(*self._args, settings=self._settings, **self._kwargs)
  File "/opt/python/lib/python3.8/site-packages/pynamodb/connection/table.py", line 238, in scan
    return self.connection.scan(
  File "/opt/python/lib/python3.8/site-packages/pynamodb/connection/base.py", line 1258, in scan
    return self.dispatch(SCAN, operation_kwargs, settings)
  File "/opt/python/lib/python3.8/site-packages/pynamodb/connection/base.py", line 329, in dispatch
    data = self._make_api_call(operation_name, operation_kwargs, settings)
  File "/opt/python/lib/python3.8/site-packages/pynamodb/connection/base.py", line 358, in _make_api_call
    request_dict = self.client._convert_to_request_dict(
```

**Comment by ikonst** [comment]
Fixed in #1083. Which version of pynamodb are you running?

**Comment by KirylKovaliov** [comment]
Thanks, we are using 5.2.1

**Comment by ikonst** [comment]
The fix is part of 5.2.2 and 5.2.3:
https://pynamodb.readthedocs.io/en/latest/release_notes.html#v5-2-3
